src/microbe_stage/PatchRegion.cs

```diff
@@ -47,17 +47,40 @@ public enum RegionType
     public int ID { get; }
 
     /// <summary>
-    ///   Regions this is next to
+    ///   Regions this is next to, and the connecting patches in the other regions
     /// </summary>
     [JsonIgnore]
-    public ISet<PatchRegion> Adjacent { get; } = new HashSet<PatchRegion>();
+    public Dictionary<PatchRegion, Patch?> Adjacent { get; } = new();
```

Comment:
Why is the connecting patch allowed to be null here? Can regions be connected so that the target patch it connects to doesn't exist? I had the impression that that is not allowed, i.e. that situation is invalid map data.

Reply:
The reason this is the way it is, is because the method that links regions together doesn't actually know the patches it is connecting to, but I think Ihave an idea on how to fix this